build_projects/shared-build-targets-utils/Utils/SharedFrameworkPublisher.cs

```diff
@@ -139,19 +139,18 @@ public void PublishSharedFramework(string outputRootDirectory, string commitHash
             PublishMutationUtilties.CleanPublishOutput(
                 sharedFrameworkNameAndVersionRoot,
                 "framework",
+                new string[] { "framework", "apphost", "hostfxr", "dotnet" },
```

Comment:
I do not think that "framework" needs to be in this array as that is the framework.deps.json whose deletion is managed by the two bools you pass in after this array.